### PR TITLE
Server: implemented delete endpoint

### PR DESCRIPTION
**Pull request by promaty**

### CODE REVIEW

File: packages/server/src/helpers/files-validator.ts [hunk 1]
```diff
@@ -1,68 +1,101 @@
...
-  static validateForCreate: publicValidateFunction = async (ctx) => {
+  static validateForCreate(ctx: Context) {
     const errors: error[] = []
     const { body: { owner }, file } = ctx.request
     if (!owner) {
       errors.push({ owner: 'Owner address must be given' })
     }
-    else if (!isAddress(owner)) {
-      errors.push({ owner: 'Given address is not valid' })
+    else if (!utils.isAddress(owner)) {
```
Comment (facuspagnuolo) [inline packages/server/src/helpers/files-validator.ts:19]: Weren't you introducing `web3-utils` in favor of `ethers` in the previous PR?

File: packages/server/src/helpers/files-validator.ts [hunk 2]
```diff
@@ -1,68 +1,101 @@
...
+      }
+    }
+  }
+
+  private static async validateSignatureAddress(ctx: Context, errors: error[]) {
+    const { signature, timestamp } = ctx.request.body
+    const { cid } = ctx.params
+    const { owner } = await FileMeta.findOne({cid})
+    const signedAddress = utils.verifyMessage(String(timestamp), signature)
+    if (owner.toLowerCase() !== signedAddress.toLowerCase()) {
```
Comment (facuspagnuolo) [inline packages/server/src/helpers/files-validator.ts:97]: The `owner` should already be lowercased right? We could probably add a sanitizer middleware for the requests content

File: packages/server/src/helpers/files-validator.ts [hunk 3]
```diff
@@ -1,68 +1,101 @@
 import { Context } from 'koa'
 import { BAD_REQUEST, NOT_FOUND } from 'http-status-codes'
-import { isAddress } from 'web3-utils'
+import { utils } from 'ethers'
 
-import { FileMeta } from '@aragonone/ipfs-background-service-shared'
+import { FileMeta, times } from '@aragonone/ipfs-background-service-shared'
 
+const SIGNATURE_TIMESTAMP_EXPIRES = times.MINUTES * 10
```
Comment (facuspagnuolo) [inline packages/server/src/helpers/files-validator.ts:7]: WDYT about `SIGNATURE_TIMESTAMP_EXPIRATION` ?

File: packages/server/src/helpers/files-validator.ts [hunk 4]
```diff
@@ -1,68 +1,101 @@
...
+
+  private static validateTimestampFormat(ctx: Context, errors: error[]) {
+    const { timestamp } = ctx.request.body
+    if (!timestamp) {
+      errors.push({timestamp: 'A timestamp must be given'})
+    }
+    else if (Number(timestamp) > Date.now()) {
+      errors.push({timestamp: 'Given timestamp is not valid'})
+    }
+    else if (Number(timestamp) < Date.now()-SIGNATURE_TIMESTAMP_EXPIRES) {
```
Comment (facuspagnuolo) [inline packages/server/src/helpers/files-validator.ts:73]: I thought this was a dash initially 😂 
```suggestion
    else if (Number(timestamp) < Date.now() - SIGNATURE_TIMESTAMP_EXPIRES) {
```

File: packages/server/src/helpers/files-validator.ts (hunk 2 above)
Comment (promaty) [reply packages/server/src/helpers/files-validator.ts]: Ah yes, I'm getting it from DB here so already lower cased. Basically all requests go directly to DB now so no need for middleware sanitizer for now.

**Pull request merged**